js-library/src/request-verifiable-presentation.ts

```diff
@@ -194,11 +194,22 @@ export const requestVerifiablePresentation = ({
   identityProvider,
 }: RequestVerifiablePresentationParams): void => {
   const handleFlowFactory = (currentFlowId: FlowId) => (evnt: MessageEvent) => {
+    // We convert the origin to URL type to avoid problems with trailing slashes
+    // when comparing it with the identityProvider.
+    let evntOriginUrl: URL | undefined;
+    try {
+      evntOriginUrl = new URL(evnt.origin);
+    } catch (err: unknown) {
+      console.warn(
+        `WARNING: expected origin to be URL, got '${evnt.origin} instead' (ignoring)`,
+      );
+      return;
+    }
     // The handler is listening to all window messages.
     // For example, a browser extension could send messages that we want to ignore.
-    if (evnt.origin !== identityProvider) {
+    if (evntOriginUrl?.origin !== identityProvider.origin) {
       console.warn(
-        `WARNING: expected origin '${identityProvider}', got '${evnt.origin}' (ignoring)`,
+        `WARNING: expected origin '${identityProvider}', got '${evntOriginUrl.origin}' (ignoring)`,
```

Comment:
Can this realistically happen? Wouldn't this be a browser bug?

Reply:
I don't know. I suspect it would be a bug, yes. But because `new URL` might fail, I added the `try/catch` just to be safe.